mtls-northbound/README.md

```diff
@@ -0,0 +1,603 @@
+
+These are the high level tasks to configure mTLS on an existing GLB configuration.
+1. Create Private CA
+   1. Create Private Certificate Authority pool, “root” CA(s) and client certificates.
+   1. Create Trust Configuration for the “root” CA(s).
+   1. Create Server TLS Policy using the Trust Configuration.
+1. Update existing GLB
+   1. Update the GLB Target HTTPS Proxy to use the Server TLS Policy.
+   1. Update the GLB Backend Service to [add custom headers](https://cloud.google.com/load-balancing/docs/https/custom-headers-global#mtls-variables) for the TLS variables.
+
+
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Have access to deploy proxies in Apigee
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * curl
+
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions in Cloud Shell. Alternatively, follow the instructions below.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=mtls-northbound/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+1. Clone the apigee-samples repo, and switch to the mtls-northbound directory
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+cd apigee-samples/mtls-northbound
+```
+
+2. Ensure you have an active GCP account selected in the Cloud shell
+
+```sh
+gcloud auth login
+```
+
+3. Edit the `env.sh` and configure the ENV vars specific to your installation.
+
+* `PROJECT` the project where your Apigee organization is located.
+* `APIGEE_HOST` the externally reachable hostname of the Apigee environment group that contains APIGEE_ENV.
+* `APIGEE_ENV` the Apigee environment where the demo resources should be created.
+* `POOL` the name of the Private CA pool (e.g. partners-pool).
+* `ROOT` the name of the Root CA in the pool (e.g. partner1-root-ca).
+* `TRUST_CONFIG` the name of the Trust Configuration for the Root CA (e.g. partner1-trust-config).
+* `CERT_NAME` the name of the client certificate used in API calls (e.g. partner-1-client-1).
+
+Other environment variables set below.
+* `TARGET_PROXY` the Target HTTPS Proxy in your GLB configuration (set below).
+* `BACKEND_SERVICE` the Backend Service in your GLB configuration (set below).
+
+Now source the `env.sh` file
+
+```bash
+source ./env.sh
+```
+
+## Create CA pool and root CAs
```

Comment:
I'm wondering if we shoud automate the following sections as part of the deploy script, and then just instruct the user to view/cat each of the files before applying them.  Thoughts?